e2e/app/page/cohort-participants-group.ts

```diff
@@ -189,6 +193,9 @@ export default class CohortParticipantsGroup {
   async deleteCriteria(criteriaName: string): Promise<void> {
     await this.clickCriteriaSnowmanIcon(criteriaName);
     await this.selectSnowmanMenu(MenuOption.DeleteCriteria);
+    // The criteria should be removed.
+    const snowmanIconLink = this.getCriteriaSnowmanIconLink(criteriaName);
+    await this.page.waitForXPath(snowmanIconLink.getXpath(), { hidden: true });
```

Comment:
Ensure `waitWhileLoading` doesn't run unless the criteria has been removed in UI.